integration_tests/run_pyspark_from_build.sh

```diff
@@ -70,6 +70,21 @@ else
         # to still launch a spark application.
         TEST_PARALLEL=`nvidia-smi --query-gpu=memory.free --format=csv,noheader | awk '{if (MAX < $1){ MAX = $1}} END {print int(MAX / (2.3 * 1024)) - 1}'`
         echo "AUTO DETECTED PARALLELISM OF $TEST_PARALLEL"
+
+        # adjust TEST_PARALLEL according to cpu cores and free memory
+        cpu_cores=$(nproc --all)
+        free_mem_mib=$(awk '/MemFree/ { printf "%d\n", $2/1024 }' /proc/meminfo)
+        max_parallel_for_cpu_cores=$(python -c "print($cpu_cores - 1)")
+        # assume 50M for each test thread
+        max_parallel_for_free_memory=$(python -c "print(int(($free_mem_mib - 1) / 50))")
```

Comment:
```suggestion
        max_parallel_for_free_memory=$(( ($free_mem_mib - 1) / 50 ))
```

Reply:
Curious about the reasoning for the `- 1` _before_ dividing by 50MB per thread.  For most free memory values, the divide will eliminate any effect of the `- 1`.

Reply:
I am also confused where the 50 MB/thread came from. I thought the default heap size for Spark is 1GiB

Reply:
"-1" is to reserve some memory for other processes in the OS.
"-1" is no need, let's only adjust 50MB.